src/org/intellij/erlang/psi/impl/ErlangPsiImplUtil.java

```diff
@@ -1839,38 +1882,91 @@ public ErlangFunctionCallParameter(@NotNull String funName, @NotNull String modu
+             myArity == getArity(resolvedFunction);
+    }
+  }
+
+  public static class ErlangFunctionCallModuleParameter extends PatternCondition<ErlangQAtom> {
+    public ErlangFunctionCallModuleParameter() {
+      super("functionCallModuleParameter");
+    }
+
+    @Override
+    public boolean accepts(@NotNull ErlangQAtom element, ProcessingContext context) {
+      ErlangFunctionCallExpression call = PsiTreeUtil.getParentOfType(element, ErlangFunctionCallExpression.class);
+      int position = call != null ? getPositionInFunctionCall(call, element) : -1;
+      if (position == -1) return false;
+
+      ErlangSpecification specification = findSpecification(resolveToFunction(call));
+      ErlangFunTypeSigs signature = specification != null ? specification.getSignature() : null;
+      if (signature == null) return false;
+
+      for (ErlangTypeSig typeSig : signature.getTypeSigList()) {
+        List<ErlangType> types = typeSig.getFunType().getFunTypeArguments().getTypeList();
+        if (types.size() <= position) continue;
+        ErlangType type = types.get(position);
+        ErlangQVar typeVar = type.getQVar();
+        if (argumentIsModule(typeVar != null ? getType(typeVar, typeSig.getTypeSigGuard()) : type)) {
+          return true;
+        }
+      }
+      return false;
+    }
+
+    @Nullable
+    private static ErlangType getType(@NotNull ErlangQVar typeVar, @Nullable ErlangTypeSigGuard sigGuard) {
+      if (sigGuard == null) return null;
+      for (ErlangTypeGuard guard : sigGuard.getTypeGuardList()) {
+        for (ErlangType topType : guard.getTypeList()) {
+          ErlangQVar var = topType.getQVar();
+          if (var != null && typeVar.getName().equals(var.getName()) && topType.getType() != null) {
+            return topType.getType();
+          }
+        }
+      }
+      return null;
+    }
+
+    private static boolean argumentIsModule(@Nullable ErlangType type) {
+      ErlangTypeRef ref = type != null ? type.getTypeRef() : null;
+      ErlangAtom atom = ref != null ? ref.getQAtom().getAtom() : null;
+      return atom != null && "module".equals(atom.getName());
+    }
+  }
+
+  public static class ErlangFunctionCallFunctionParameter extends PatternCondition<ErlangQAtom> {
+    public ErlangFunctionCallFunctionParameter() {
+      super("functionCallFunctionParameter");
+    }
+
+    @Override
+    public boolean accepts(@NotNull ErlangQAtom element, ProcessingContext context) {
+      ErlangFunctionCallExpression call = PsiTreeUtil.getParentOfType(element, ErlangFunctionCallExpression.class);
+      if (call == null) return false;
+      int position = getPositionInFunctionCall(call, element);
+      List<ErlangExpression> expressions = call.getArgumentList().getExpressionList();
+
+      return position > 0 && isModule(expressions.get(position - 1), context) &&
+             position + 1 < expressions.size() && expressions.get(position + 1) instanceof ErlangListExpression;
```

Comment:
This instanceof condition is wrong - it should check if expected parameter's type is a list instead, or you can drop this check altogether.